src/event/quic/ngx_event_quic_streams.c

```diff
@@ -201,6 +201,16 @@ ngx_quic_close_streams(ngx_connection_t *c, ngx_quic_connection_t *qc)
 
     while (node) {
         qs = (ngx_quic_stream_t *) node;
+
+        if (qs->destroyed) {
+            if (tree->root == tree->sentinel) {
+                return NGX_OK;
+            }
+
+            node = ngx_rbtree_min(tree->root, tree->sentinel);
```

Comment:
This may not give the actual next node, but the first one which was not closed. In this case for all such non-closed nodes read handler will be called again.

Reply:
Indeed, this is seemingly possible for stream requests blocked on AIO, in which case termination doesn't happen immediately.
On the other hand, I don't see how this differs from QUIC close handler posted or scheduled to resume QUIC connection close. In this case, stream read handlers for non-closed nodes are called again as well. This is arguably another bug, ngx_quic_close_streams() should be placed under !qc->closing.

Reply:
On a related note, see also 9a3ec2023 where calling read handlers again already happens to facilitate HTTP/2 connection draining in lingering, and I assume this will always happen after converting ngx_quic_close_streams() to posted events.
The question is why calling handlers again may be bad.